pyquil/tests/test_gates.py

```diff
@@ -103,10 +103,27 @@ def test_dagger_gate():
     g = X(0).dagger().dagger() == "DAGGER DAGGER X 0"
 
 
+def test_forked_gate():
+    g = RX(0.0, 0).forked(1, [1.0])
+    assert g.out() == "FORKED RX(0,1.0) 1 0"
+    g = RX(0.0, 0).forked(1, [1.0]) \
+                    .forked(2, [2.0, 3.0])
+    assert g.out() == "FORKED FORKED RX(0,1.0,2.0,3.0) 2 1 0"
```

Comment:
It's not hard to image a future in which these text-based assertions cause serious headache. For example, the compiler pretty prints parameter lists and inserts spaces after commas.

Reply:
I agree. Fixing all of these tests is a low hanging fruit worthy of a separate PR.